analyze/analyze.rs

```diff
@@ -907,16 +907,55 @@ impl PartialOrd for MonosEntry {
 
 impl Ord for MonosEntry {
     fn cmp(&self, rhs: &MonosEntry) -> cmp::Ordering {
-        rhs.approx_potential_savings
-            .cmp(&self.approx_potential_savings)
+        rhs.bloat
+            .cmp(&self.bloat)
+            .then(rhs.size.cmp(&self.size))
             .then(self.insts.cmp(&rhs.insts))
-            .then(self.generic.cmp(&rhs.generic))
+            .then(self.name.cmp(&rhs.name))
     }
 }
 
 impl traits::Emit for Monos {
     #[cfg(feature = "emit_text")]
     fn emit_text(&self, items: &ir::Items, dest: &mut io::Write) -> Result<(), traits::Error> {
+        struct TableRow {
+            bloat: Option<u32>,
+            bloat_percent: Option<f64>,
+            size: u32,
+            size_percent: f64,
+            name: String,
+        };
+
+        // Given an entry representing a generic function and its various
+        // monomorphizations, return a vector of table rows.
+        fn process_entry(entry: &MonosEntry, total_size: f64) -> Vec<TableRow> {
+            let MonosEntry {
+                name,
+                insts,
+                size,
+                bloat,
+            } = entry;
+
+            let get_size_percent = |x: u32| f64::from(x) / total_size * 100.0;
+
+            let mut rows = vec![TableRow {
```

Comment:
I think this function could return `-> impl Iterator<Item = TableRow>` and avoid separate heap allocations for each vec. Not a big deal, though.

Reply:
Ooh neat! So, I *think* I took care of this, but I'm not completely sure? I'll hold off on merging this until you have a chance to check on the revised version, but I'd like to try and do it the way you mentioned :)